src/main/scala/com/twitter/finagle/postgres/Client.scala

```diff
@@ -81,6 +81,8 @@ class Client(factory: ServiceFactory[PgRequest, PgResponse], id:String) {
       service =>
         parse(sql, Some(service)).map { name =>
           new PreparedStatementImpl(name, service)
+        }.onFailure {
+          err => service.close()
```

Comment:
The `PreparedStatementImpl` needs to use the same service that `Parse` was executed on (because that session is where the prepared statement lives)